Block/AccountBlockService.php

```diff
@@ -30,21 +31,27 @@
 class AccountBlockService extends BaseBlockService
 {
     /**
-     * @var TokenStorageInterface
+     * @var TokenStorageInterface|SecurityContextInterface
      */
     private $tokenStorage;
 
     /**
      * Constructor.
      *
-     * @param string                $name
-     * @param EngineInterface       $templating
-     * @param TokenStorageInterface $tokenStorage
+     * NEXT_MAJOR: Go back to signature class check when bumping requirements to SF 2.6+.
+     * 
+     * @param string                                         $name
+     * @param EngineInterface                                $templating
+     * @param TokenStorageInterface|SecurityContextInterface $tokenStorage
      */
-    public function __construct($name, EngineInterface $templating, TokenStorageInterface $tokenStorage)
+    public function __construct($name, EngineInterface $templating, $tokenStorage)
     {
         parent::__construct($name, $templating);
 
+        if (!$tokenStorage instanceof TokenStorageInterface && !$tokenStorage instanceof SecurityContextInterface) {
+            throw new \InvalidArgumentException('Argument 1 should be an instance of Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface or Symfony\Component\Security\Core\SecurityContextInterface');
```

Comment:
Not telling you to split the string since this is still [debatable](https://github.com/sonata-project/dev-kit/issues/218), but I think you should add least but the message on its own line cc @sagikazarmark 

```php
throw new \InvalidArgumentException(
    'Argument 1 should yadda yadda…'
);
```